### PR TITLE
Remove aggregate panic

### PR DESCRIPTION
**Pull request by fpetkovski**
We currently panic in the hash aggregate function when an operation
is not supported.
    
This commit removes the panic and replaces it with an error, so that we can fall back to the old engine. 
I had to refactor the codebase a bit in order to enable the change.

### CODE REVIEW

**Review by saswatamcode — APPROVED** [review]
LGTM!

File: physicalplan/aggregate/scalar_table.go [hunk 1]
```diff
@@ -118,81 +112,100 @@ type accumulator struct {
...
-				value = 0
-			},
+		return func() *accumulator {
+			var value float64
+			var hasValue bool
+
+			return &accumulator{
+				AddFunc: func(v float64) {
+					hasValue = true
+					value = math.Max(value, v)
```
Comment (yeya24) [inline physicalplan/aggregate/scalar_table.go:144]: We need to check `hasValue` in case v < 0 ? Or initialize value as the min float64

Comment (fpetkovski) [reply physicalplan/aggregate/scalar_table.go:144]: I think you're right. Will create an issue for this 👍

Comment (fpetkovski) [reply physicalplan/aggregate/scalar_table.go:144]: Created https://github.com/thanos-community/promql-engine/issues/37

**Pull request merged**